python-lib/dku_deeplearning_image/utils.py

```diff
@@ -77,25 +78,39 @@ def can_use_gpu():
     return len(tf.config.experimental.list_physical_devices('GPU')) > 0
 
 
-def set_gpu_options(should_use_gpu, gpu_list, memory_limit):
+def set_gpu_options(should_use_gpu, gpu_list, gpu_memory_allocation_mode, memory_limit_ratio=None):
     log_info("load_gpu_options")
     if should_use_gpu and can_use_gpu():
         log_info("should use GPU")
         gpus = tf.config.experimental.list_physical_devices('GPU')
+        for i, g in enumerate(gpus):
+            g.id = i
         gpus_to_use = [gpus[int(i)] for i in gpu_list] or gpus
-        if memory_limit:
+        log_info("GPUs on the machine: {}".format(str([g.id for g in GPUtil.getGPUs()])))
```

Comment:
You don't need to `str`, format takes care of that for you. Bonus: you could use f"" string formatting

Reply:
I am always a little bit scared of using f string syntax, but it works for all Python 3 versions right?